File: grouped-by-commas/groupedByCommas.js
const groupByCommas = (number) => {
    const string = number.toString();
    if(string.length <= 3) return string;
    
    const split = string.split('').reverse();
    let count = 1;
    split.forEach((element, index) => {
        if(index === 3){
            split.splice(index, 0, ',');
        }
        else if(index % 3 === 0 && index !== 0){
            split.splice(index + count, 0, ',');
            count++;
        };
    });
    
    return split.reverse().join('');
};

module.exports = { groupByCommas };

//https://www.codewars.com/kata/5274e122fc75c0943d000148/train/javascript

File: python-venv/grouped_by_commas/kata_function/grouped_by_commas.py
def group_by_commas(number):

    string = str(number)
    if len(string) <= 3:
        return string
    
    split = list(string)
    split.reverse()
    count = 1

    for i in range(0, len(string), 1):
        if i == 3:
            split.insert(i, ',')
        elif i != 0 and i % 3 == 0:
            split.insert(i + count, ',')
            count = count + 1

    split.reverse()
    return "".join(split)

#https://www.codewars.com/kata/5274e122fc75c0943d000148/train/python